### PR TITLE
dirty_memory_manager: tidy up

### PR DESCRIPTION
**Pull request by avikivity**
A collection of small cleanups, and a bug fix.

### CODE REVIEW

File: dirty_memory_manager.hh [hunk 1]
```diff
@@ -94,14 +94,16 @@ struct reclaim_config {
     reclaim_stop_callback stop_reclaiming = [] () noexcept {};
 };
 
-class allocation_queue {
+// Groups regions for the purpose of statistics.  Can be nested.
```
Comment (denesb) [inline dirty_memory_manager.hh:97]: "Can be nested" should be removed. Can be a follow-up.

File: dirty_memory_manager.cc [hunk 2]
```diff
@@ -107,25 +108,23 @@ region_group::start_releaser(scheduling_group deferred_work_sg) {
...
-
+        {
+            while (!_shutdown_requested) {
                 if (!_blocked_requests.empty() && execution_permitted()) {
                     execute_one();
-                    return make_ready_future<stop_iteration>(stop_iteration::no);
+                    co_await coroutine::maybe_yield();
                 } else {
+                  // We want `rl` to hold for the call to _relief.wait(), but not to wait
+                  // for the future to resolve, hence the inner lambda.
```
Comment (denesb) [inline dirty_memory_manager.cc]: Can also be done without a lambda:
```c++
future<> fut = make_ready_future<>();
{
    tracker_reclaimer_lock rl(logalloc::shard_tracker());
    fut = _relief.wait();
}
co_await std::move(fut);
```

**Review by denesb — APPROVED** [review]

**Comment by denesb** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/releng/job/Scylla-CI/2601/

There are 686 failed tests. Either you broke something fundamental or CI broke down.

File: dirty_memory_manager.hh (hunk 1 above)
Comment (avikivity) [reply dirty_memory_manager.hh:97]: Since it's a collection of cleanups, I'll add this one too.

File: dirty_memory_manager.cc (hunk 2 above)
Comment (avikivity) [reply dirty_memory_manager.cc]: I like the lambda better since it avoids the artificial variable.

**Comment by avikivity** [comment]
v2:
 - new patch "dirty_memory_manager: move region_group::_releaser after _shutdown_requested" fixes debug mode failures due to use-before-init
 - update comment (new patch, at end)
 - move region_group data members to top-of-class (new patch, at end)

**Review by denesb — APPROVED** [review]

**Pull request merged**